code/go/0chain.net/chaincore/httpclientutil/http_client_util_test.go

```diff
@@ -0,0 +1,1050 @@
+		args    args
+		want    *http.Request
+		wantErr bool
+	}{
+		{
+			name: "OK",
+			args: args{
+				method: "",
+				url:    url,
+				data:   data,
+				ID:     id,
+				pkey:   pKey,
+			},
+			want:    req,
+			wantErr: false,
+		},
+		{
+			name: "ERR",
+			args: args{
+				url: string(rune(0x7f)),
+			},
+			want:    req,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			got, err := NewHTTPRequest(tt.args.method, tt.args.url, tt.args.data, tt.args.ID, tt.args.pkey)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("NewHTTPRequest() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != nil && !tt.wantErr {
+				assert.Equal(t, tt.want.URL, got.URL)
+				tt.want.URL = nil
+				got.URL = nil
+
+				assert.Equal(t, tt.want.Body, got.Body)
+			}
+		})
+	}
+}
+
+func TestSendPostRequest(t *testing.T) {
+	t.Skip("need to protect global variable httpClient against concurrent access")
+
+	t.Parallel()
+
+	server := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+			},
+		),
+	)
+	defer server.Close()
+
+	type args struct {
+		url  string
+		data []byte
+		ID   string
+		pkey string
+		wg   *sync.WaitGroup
+	}
+	tests := []struct {
+		name    string
+		args    args
+		client  *http.Client
+		want    []byte
+		wantErr bool
+	}{
+		{
+			name: "Request_Creating_ERR",
+			args: args{
+				url: string(rune(0x7f)),
+			},
+			wantErr: true,
+		},
+		{
+			name:   "Client_ERR",
+			client: server.Client(),
+			args: args{
+				url: "/",
+			},
+			wantErr: true,
+		},
+		{
+			name:   "OK",
+			client: server.Client(),
+			args: args{
+				url: server.URL,
+			},
+			want:    make([]byte, 0, 512),
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			httpClient = tt.client
+
+			got, err := SendPostRequest(tt.args.url, tt.args.data, tt.args.ID, tt.args.pkey, tt.args.wg)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("SendPostRequest() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("SendPostRequest() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestSendMultiPostRequest(t *testing.T) {
+	t.Skip("need to protect global variable httpClient against concurrent access")
+
+	t.Parallel()
+
+	server := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+			},
+		),
+	)
+	defer server.Close()
+
+	type args struct {
+		urls []string
+		data []byte
+		ID   string
+		pkey string
+	}
+	tests := []struct {
+		name   string
+		client *http.Client
+		args   args
+	}{
+		{
+			name:   "OK",
+			client: server.Client(),
+			args: args{
+				urls: []string{
+					server.URL,
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			httpClient = tt.client
```

Comment:
Don't redfine this its effectivly a constant.

Reply:
Removed.